File: apps/desktop/src/types/quote.ts
import type { Market, WatchItemType } from './watchItem';

export type QuoteStatus = 'normal' | 'delay' | 'closed' | 'error';

export type Quote = {
  symbol: string;
  market: Market;
  type: WatchItemType;
  name: string;
  price: number | null;
  changeAmount: number | null;
  changePercent: number | null;
  volume?: number | null;
  amount?: number | null;
  source: string;
  status: QuoteStatus;
  updatedAt: string;
  cached?: boolean;
};

export type DisplayMode =
  | 'excel'
  | 'weather'
  | 'calendar'
  | 'ledger'
  | 'project'
  | 'mail'
  | 'logistics'
  | 'monitor';

export type DisplayColumn = {
  field: string;
  headerName: string;
  width?: number;
  align?: 'left' | 'right' | 'center';
};

export type DisplayRow = Record<string, string | number | null | undefined>;

File: services/quote-gateway/app/schemas/quote.py
from typing import Literal

from pydantic import BaseModel


Market = Literal["SH", "SZ", "BJ", "HK", "US", "FUND", "INDEX"]
WatchItemType = Literal["stock", "fund", "etf", "index"]
QuoteStatus = Literal["normal", "delay", "closed", "error"]


class Quote(BaseModel):
    symbol: str
    market: Market
    type: WatchItemType
    name: str
    price: float | None
    changeAmount: float | None
    changePercent: float | None
    volume: float | None = None
    amount: float | None = None
    source: str
    status: QuoteStatus
    updatedAt: str
    cached: bool = False
